APPLICATION DATA

Nos encontramos ante un problema de aprendizaje supervisado, ya que contamos con una variable objetivo, TARGET, que clasifica a los clientes según si pagan su préstamo (valor 0) o no (valor 1). En este análisis exploratorio de datos (EDA) investigaremos el dataset application_data.csv para comprender las características y comportamientos de las variables, identificando patrones, distribuciones y relaciones clave. Además, abordaremos posibles valores atípicos, sesgos y valores faltantes que puedan afectar el rendimiento de futuros modelos predictivos.

Este análisis tiene como objetivo principal ayudar al banco a tomar decisiones más informadas, reduciendo el riesgo de impagos al identificar perfiles de clientes con mayor probabilidad de incumplimiento, sin perjudicar a quienes cumplen; a la vez que también reduce la probabilidad de dejar pasar a clientes que no incumplirán. El modelo se basará en datos proporcionados por el cliente al solicitar el préstamo, como información demográfica, financiera y laboral, y será evaluado mediante métricas clave como la curva ROC, AUC, precisión y recall.

Los objetivos específicos de este EDA son:

- Analizar las distribuciones de las variables categóricas y numéricas, explorando su relación con la variable objetivo.
- Identificar y tratar valores nulos o atípicos para mejorar la calidad de los datos.
- Explorar relaciones clave entre las variables y su impacto en el objetivo.
- Proponer transformaciones y selecciones de variables para optimizar el dataset para el modelado.



In [2]:
import pandas as pd
import seaborn as sb
import numpy as np
from matplotlib import pyplot as pyplot
import plotly.express as px

pd.set_option('display.max_columns', 10000)
pd.set_option('display.max_rows', 10000)
pd.set_option('display.width', 10000)

FUNCIONES

In [3]:
def dame_variables_categoricas(dataset=None):
    if dataset is None:
        print(u'\nFaltan argumentos por pasar a la función')
        return 1
    lista_variables_categoricas = []
    other = []
    for i in dataset.columns:
        if (dataset[i].dtype!=float) & (dataset[i].dtype!=int):
            unicos = int(len(np.unique(dataset[i].dropna(axis=0, how='all'))))
            if unicos < 100:
                lista_variables_categoricas.append(i)
            else:
                other.append(i)

    return lista_variables_categoricas, other

In [4]:
path_folder = "../data/"
pd_loan = pd.read_csv(path_folder + "application_data.csv", low_memory = False)

pd_loan.head()

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  AMT_GOODS_PRICE NAME_TYPE_SUITE NAME_INCOME_TYPE            NAME_EDUCATION_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  REGION_POPULATION_RELATIVE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  DAYS_ID_PUBLISH  OWN_CAR_AGE  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  FLAG_CONT_MOBILE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  CNT_FAM_MEMBERS  REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY WEEKDAY_APPR_PROCESS_START  HOUR_APPR_PROCESS_START  REG_REGION_NOT_LIVE_REGION  REG_REGION_NOT_WORK_REGION  LIVE_REGION_NOT_WORK_REGION  REG_CITY_NOT_LIVE_CITY  REG_CITY_NOT_WORK_CITY  LIVE_CITY_NOT_WORK_CITY       ORGANIZATION_TYPE  EXT_SOURCE_1  EXT_SOURCE_2  EXT_SOURCE_3  APARTMENTS_AVG  BASEMENTAREA_AVG  YEARS_BEGINEXPLUATATION_AVG  YEARS_BUILD_AVG  COMMONAREA_AVG  ELEVATORS_AVG  ENTRANCES_AVG  FLOORSMAX_AVG  FLOORSMIN_AVG  LANDAREA_AVG  LIVINGAPARTMENTS_AVG  LIVINGAREA_AVG  NONLIVINGAPARTMENTS_AVG  NONLIVINGAREA_AVG  APARTMENTS_MODE  BASEMENTAREA_MODE  YEARS_BEGINEXPLUATATION_MODE  YEARS_BUILD_MODE  COMMONAREA_MODE  ELEVATORS_MODE  ENTRANCES_MODE  FLOORSMAX_MODE  FLOORSMIN_MODE  LANDAREA_MODE  LIVINGAPARTMENTS_MODE  LIVINGAREA_MODE  NONLIVINGAPARTMENTS_MODE  NONLIVINGAREA_MODE  APARTMENTS_MEDI  BASEMENTAREA_MEDI  YEARS_BEGINEXPLUATATION_MEDI  YEARS_BUILD_MEDI  COMMONAREA_MEDI  ELEVATORS_MEDI  ENTRANCES_MEDI  FLOORSMAX_MEDI  FLOORSMIN_MEDI  LANDAREA_MEDI  LIVINGAPARTMENTS_MEDI  LIVINGAREA_MEDI  NONLIVINGAPARTMENTS_MEDI  NONLIVINGAREA_MEDI FONDKAPREMONT_MODE  HOUSETYPE_MODE  TOTALAREA_MODE WALLSMATERIAL_MODE EMERGENCYSTATE_MODE  OBS_30_CNT_SOCIAL_CIRCLE  DEF_30_CNT_SOCIAL_CIRCLE  OBS_60_CNT_SOCIAL_CIRCLE  DEF_60_CNT_SOCIAL_CIRCLE  DAYS_LAST_PHONE_CHANGE  FLAG_DOCUMENT_2  FLAG_DOCUMENT_3  FLAG_DOCUMENT_4  FLAG_DOCUMENT_5  FLAG_DOCUMENT_6  FLAG_DOCUMENT_7  FLAG_DOCUMENT_8  FLAG_DOCUMENT_9  FLAG_DOCUMENT_10  FLAG_DOCUMENT_11  FLAG_DOCUMENT_12  FLAG_DOCUMENT_13  FLAG_DOCUMENT_14  FLAG_DOCUMENT_15  FLAG_DOCUMENT_16  FLAG_DOCUMENT_17  FLAG_DOCUMENT_18  FLAG_DOCUMENT_19  FLAG_DOCUMENT_20  FLAG_DOCUMENT_21  AMT_REQ_CREDIT_BUREAU_HOUR  AMT_REQ_CREDIT_BUREAU_DAY  AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  AMT_REQ_CREDIT_BUREAU_QRT  AMT_REQ_CREDIT_BUREAU_YEAR
0      100002       1         Cash loans           M            N               Y             0          202500.0    406597.5      24700.5         351000.0   Unaccompanied          Working  Secondary / secondary special  Single / not married  House / apartment                    0.018801       -9461           -637            -3648.0            -2120          NaN           1               1                0                 1           1           0        Laborers              1.0                     2                            2                  WEDNESDAY                       10                           0                           0                            0                       0                       0                        0  Business Entity Type 3      0.083037      0.262949      0.139376          0.0247            0.0369                       0.9722           0.6192          0.0143           0.00         0.0690         0.0833         0.1250        0.0369                0.0202          0.0190                   0.0000             0.0000           0.0252             0.0383                        0.9722            0.6341           0.0144          0.0000          0.0690          0.0833          0.1250         0.0377                  0.022           0.0198                       0.0                 0.0           0.0250             0.0369                        0.9722            0.6243           0.0144            0.00          0.0690          0.0833          0.1250         0.0375                 0.0205           0.0193                    0.0000                0.00   reg oper account  block of flats          0.0149       Stone, brick       

ANÁLISIS GENERAL DE LA TABLA Y PROCESAMIENTO INICIAL

In [5]:
print(pd_loan.shape, pd_loan.drop_duplicates().shape)

(307511, 122) (307511, 122)


Nos encontramos ante un dataframe de 307,511 filas y 122 columnas

In [6]:

pd_loan.dtypes.to_dict()

{'SK_ID_CURR': dtype('int64'),
 'TARGET': dtype('int64'),
 'NAME_CONTRACT_TYPE': dtype('O'),
 'CODE_GENDER': dtype('O'),
 'FLAG_OWN_CAR': dtype('O'),
 'FLAG_OWN_REALTY': dtype('O'),
 'CNT_CHILDREN': dtype('int64'),
 'AMT_INCOME_TOTAL': dtype('float64'),
 'AMT_CREDIT': dtype('float64'),
 'AMT_ANNUITY': dtype('float64'),
 'AMT_GOODS_PRICE': dtype('float64'),
 'NAME_TYPE_SUITE': dtype('O'),
 'NAME_INCOME_TYPE': dtype('O'),
 'NAME_EDUCATION_TYPE': dtype('O'),
 'NAME_FAMILY_STATUS': dtype('O'),
 'NAME_HOUSING_TYPE': dtype('O'),
 'REGION_POPULATION_RELATIVE': dtype('float64'),
 'DAYS_BIRTH': dtype('int64'),
 'DAYS_EMPLOYED': dtype('int64'),
 'DAYS_REGISTRATION': dtype('float64'),
 'DAYS_ID_PUBLISH': dtype('int64'),
 'OWN_CAR_AGE': dtype('float64'),
 'FLAG_MOBIL': dtype('int64'),
 'FLAG_EMP_PHONE': dtype('int64'),
 'FLAG_WORK_PHONE': dtype('int64'),
 'FLAG_CONT_MOBILE': dtype('int64'),
 'FLAG_PHONE': dtype('int64'),
 'FLAG_EMAIL': dtype('int64'),
 'OCCUPATION_TYPE': dtype('O'),
 'CNT_FAM_MEMB

EXPLORACIÓN DE VARIABLE OBJETIVO Y TRATAMIENTO

In [7]:
pd_plot_target = pd_loan['TARGET']\
        .value_counts(normalize=True)\
        .mul(100).rename('percent').reset_index()

pd_plot_target_conteo = pd_loan['TARGET'].value_counts().reset_index()
pd_plot_target_pc = pd.merge(pd_plot_target, pd_plot_target_conteo, on = ['TARGET'], how='inner')

print(pd_plot_target_pc)

   TARGET    percent   count
0       0  91.927118  282686
1       1   8.072882   24825


Nuestra variable objetivo "TARGET" puede tomar valores 0 y 1. Contamos con 282,686 filas que cuentan con el valor 0 de esta variable (un 92% del dataset) y 24,825 cuentan con el valor 1 (8% del dataset)

In [8]:
fig = px.histogram(pd_plot_target_pc, x="TARGET", y=['percent'])
fig.update_xaxes(tickvals=[0, 1])
fig.show()

Aquí podemos visualizar la distribución de los valores de la variable TARGET en suma de valor, donde se ve que más del 90% del dataset toma el valor 0, por lo que no impagan por norma general. 

In [9]:
pd_series_null_columns = pd_loan.isnull().sum().sort_values(ascending=False)
pd_series_null_rows = pd_loan.isnull().sum(axis=1).sort_values(ascending=False)
print(pd_series_null_columns.shape, pd_series_null_rows.shape)

pd_null_columnas = pd.DataFrame(pd_series_null_columns, columns=['nulos_columnas'])     
pd_null_filas = pd.DataFrame(pd_series_null_rows, columns=['nulos_filas'])  
pd_null_filas['TARGET'] = pd_loan['TARGET'].copy()
pd_null_columnas['porcentaje_columnas'] = pd_null_columnas['nulos_columnas']/pd_loan.shape[0]
pd_null_filas['porcentaje_filas']= pd_null_filas['nulos_filas']/pd_loan.shape[1]

(122,) (307511,)


In [10]:
pd_null_columnas

nulos_columnas  porcentaje_columnas
COMMONAREA_AVG                        214865             0.698723
COMMONAREA_MODE                       214865             0.698723
COMMONAREA_MEDI                       214865             0.698723
NONLIVINGAPARTMENTS_MEDI              213514             0.694330
NONLIVINGAPARTMENTS_MODE              213514             0.694330
NONLIVINGAPARTMENTS_AVG               213514             0.694330
FONDKAPREMONT_MODE                    210295             0.683862
LIVINGAPARTMENTS_AVG                  210199             0.683550
LIVINGAPARTMENTS_MEDI                 210199             0.683550
LIVINGAPARTMENTS_MODE                 210199             0.683550
FLOORSMIN_MODE                        208642             0.678486
FLOORSMIN_AVG                         208642             0.678486
FLOORSMIN_MEDI                        208642             0.678486
YEARS_BUILD_AVG                       204488             0.664978
YEARS_BUILD_MODE                      204488             0.664978
YEARS_BUILD_MEDI                      204488             0.664978
OWN_CAR_AGE                           202929             0.659908
LANDAREA_MEDI                         182590             0.593767
LANDAREA_AVG                          182590             0.593767
LANDAREA_MODE                         182590             0.593767
BASEMENTAREA_MODE                     179943             0.585160
BASEMENTAREA_MEDI                     179943             0.585160
BASEMENTAREA_AVG                      179943             0.585160
EXT_SOURCE_1                          173378             0.563811
NONLIVINGAREA_MODE                    169682             0.551792
NONLIVINGAREA_AVG                     169682             0.551792
NONLIVINGAREA_MEDI                    169682             0.551792
ELEVATORS_AVG                         163891             0.532960
ELEVATORS_MEDI                        163891             0.532960
ELEVATORS_MODE                        163891             0.532960
WALLSMATERIAL_MODE                    156341             0.508408
APARTMENTS_AVG                        156061             0.507497
APARTMENTS_MEDI                       156061             0.507497
APARTMENTS_MODE                       156061             0.507497
ENTRANCES_MODE                        154828             0.503488
ENTRANCES_MEDI                        154828             0.503488
ENTRANCES_AVG                         154828             0.503488
LIVINGAREA_AVG                        154350             0.501933
LIVINGAREA_MEDI                       154350             0.501933
LIVINGAREA_MODE                       154350             0.501933
HOUSETYPE_MODE                        154297             0.501761
FLOORSMAX_MODE                        153020             0.497608
FLOORSMAX_AVG                         153020             0.497608
FLOORSMAX_MEDI                        153020             0.497608
YEARS_BEGINEXPLUATATION_MODE          150007             0.487810
YEARS_BEGINEXPLUATATION_MEDI          150007             0.487810
YEARS_BEGINEXPLUATATION_AVG           150007             0.487810
TOTALAREA_MODE                        148431             0.482685
EMERGENCYSTATE_MODE                   145755             0.473983
OCCUPATION_TYPE                        96391             0.313455
EXT_SOURCE_3                           60965             0.198253
AMT_REQ_CREDIT_BUREAU_HOUR             41519             0.135016
AMT_REQ_CREDIT_BUREAU_WEEK             41519             0.135016
AMT_REQ_CREDIT_BUREAU_MON              41519             0.135016
AMT_REQ_CREDIT_BUREAU_YEAR             41519             0.135016
AMT_REQ_CREDIT_BUREAU_DAY              41519             0.135016
AMT_REQ_CREDIT_BUREAU_QRT              41519             0.135016
NAME_TYPE_SUITE                         1292             0.004201
DEF_60_CNT_SOCIAL_CIRCLE                1021             0.003320
OBS_60_CNT_SOCIAL_CIRCLE                1021             0.003320
DEF_

In [11]:
threshold=0.8
list_vars_not_null = list(pd_null_columnas[pd_null_columnas['porcentaje_columnas']<threshold].index)
pd_loan_filter_null = pd_loan.loc[:, list_vars_not_null]
pd_loan_filter_null.shape

(307511, 122)

Hemos determinado que trabajaremos con aquellas columnas que tengan menos de un 80% de valores nulos, por lo que nos quedaremos con las 122 columnas, aunque si vemos en un futuro que no son necesarias ciertas columnas con tantos valores nulos, podríamos imputarlas o eliminarlas.

In [12]:
pd_null_filas

nulos_filas  TARGET  porcentaje_filas
244833           61       0               0.5
269786           61       0               0.5
69707            61       0               0.5
116937           61       0               0.5
185713           61       0               0.5
...             ...     ...               ...
133082            0       0               0.0
104165            0       0               0.0
104172            0       0               0.0
104120            0       0               0.0
180398            0       0               0.0

[307511 rows x 3 columns]

TIPOS: VARIABLES CATEGÓRICAS Y NUMÉRICAS

In [13]:
list_cat_vars, other = dame_variables_categoricas(dataset=pd_loan_filter_null)
pd_loan_filter_null[list_cat_vars] = pd_loan_filter_null[list_cat_vars].astype("category")
pd_loan_filter_null[list_cat_vars].head()

FONDKAPREMONT_MODE WALLSMATERIAL_MODE  HOUSETYPE_MODE EMERGENCYSTATE_MODE OCCUPATION_TYPE NAME_TYPE_SUITE       ORGANIZATION_TYPE NAME_CONTRACT_TYPE FLAG_OWN_CAR NAME_INCOME_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE            NAME_EDUCATION_TYPE CODE_GENDER FLAG_OWN_REALTY WEEKDAY_APPR_PROCESS_START
0   reg oper account       Stone, brick  block of flats                  No        Laborers   Unaccompanied  Business Entity Type 3         Cash loans            N          Working  Single / not married  House / apartment  Secondary / secondary special           M               Y                  WEDNESDAY
1   reg oper account              Block  block of flats                  No      Core staff          Family                  School         Cash loans            N    State servant               Married  House / apartment               Higher education           F               N                     MONDAY
2                NaN                NaN             NaN                 NaN        Laborers   Unaccompanied              Government    Revolving loans            Y          Working  Single / not married  House / apartment  Secondary / secondary special           M               Y                     MONDAY
3                NaN                NaN             NaN                 NaN        Laborers   Unaccompanied  Business Entity Type 3         Cash loans            N          Working        Civil marriage  House / apartment  Secondary / secondary special           F               Y                  WEDNESDAY
4                NaN                NaN             NaN                 NaN      Core staff   Unaccompanied                Religion         Cash loans            N          Working  Single / not married  House / apartment  Secondary / secondary special           M               Y                   THURSDAY

In [14]:
list_cat_vars

['FONDKAPREMONT_MODE',
 'WALLSMATERIAL_MODE',
 'HOUSETYPE_MODE',
 'EMERGENCYSTATE_MODE',
 'OCCUPATION_TYPE',
 'NAME_TYPE_SUITE',
 'ORGANIZATION_TYPE',
 'NAME_CONTRACT_TYPE',
 'FLAG_OWN_CAR',
 'NAME_INCOME_TYPE',
 'NAME_FAMILY_STATUS',
 'NAME_HOUSING_TYPE',
 'NAME_EDUCATION_TYPE',
 'CODE_GENDER',
 'FLAG_OWN_REALTY',
 'WEEKDAY_APPR_PROCESS_START']

Tenemos 11 columnas de variables categóricas

In [15]:
pd_loan_filter_null['AMT_INCOME_TOTAL'].value_counts()

AMT_INCOME_TOTAL
1.350000e+05    35750
1.125000e+05    31019
1.575000e+05    26556
1.800000e+05    24719
9.000000e+04    22483
2.250000e+05    20698
2.025000e+05    16341
6.750000e+04    11147
2.700000e+05    10827
8.100000e+04     6001
3.150000e+05     5862
2.475000e+05     5344
1.260000e+05     4668
3.600000e+05     4115
1.215000e+05     3848
9.900000e+04     3633
2.925000e+05     3418
1.035000e+05     3330
7.650000e+04     3146
1.440000e+05     3123
1.710000e+05     2647
4.500000e+05     2435
1.170000e+05     2390
5.400000e+04     2319
7.200000e+04     2187
1.665000e+05     2053
4.500000e+04     2043
1.080000e+05     1946
3.375000e+05     1790
9.450000e+04     1747
1.620000e+05     1702
1.890000e+05     1691
4.050000e+05     1660
8.550000e+04     1514
1.485000e+05     1486
2.160000e+05     1406
1.305000e+05     1256
5.850000e+04     1182
6.300000e+04     1145
1.935000e+05     1061
1.530000e+05     1012
2.115000e+05      985
3.825000e+05      966
1.395000e+05      834
1.755000e+05   

In [16]:
pd_loan_filter_null['AMT_CREDIT'].value_counts()

AMT_CREDIT
450000.0     9709
675000.0     8877
225000.0     8162
180000.0     7342
270000.0     7241
900000.0     6246
254700.0     4500
545040.0     4437
808650.0     4152
135000.0     3660
755190.0     3524
1125000.0    3493
521280.0     3175
284400.0     3096
1078200.0    2856
454500.0     2649
1288350.0    2613
269550.0     2463
202500.0     2339
781920.0     2328
1350000.0    2321
239850.0     2084
315000.0     2072
247500.0     2057
640080.0     2041
1006920.0    1883
540000.0     1821
360000.0     1743
508495.5     1717
592560.0     1688
497520.0     1665
405000.0     1661
835380.0     1600
337500.0     1588
679500.0     1555
942300.0     1431
314100.0     1394
152820.0     1366
1546020.0    1279
495000.0     1268
157500.0     1203
585000.0     1200
728460.0     1065
526491.0     1012
544491.0     1009
343800.0      939
1223010.0     930
312768.0      929
247275.0      920
1800000.0     900
814041.0      897
630000.0      895
490495.5      842
101880.0      841
263686.5      827

In [17]:
pd_loan_filter_null[list_cat_vars].dtypes
print(pd_loan_filter_null[list_cat_vars])

       FONDKAPREMONT_MODE WALLSMATERIAL_MODE  HOUSETYPE_MODE EMERGENCYSTATE_MODE OCCUPATION_TYPE NAME_TYPE_SUITE       ORGANIZATION_TYPE NAME_CONTRACT_TYPE FLAG_OWN_CAR      NAME_INCOME_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE            NAME_EDUCATION_TYPE CODE_GENDER FLAG_OWN_REALTY WEEKDAY_APPR_PROCESS_START
0        reg oper account       Stone, brick  block of flats                  No        Laborers   Unaccompanied  Business Entity Type 3         Cash loans            N               Working  Single / not married  House / apartment  Secondary / secondary special           M               Y                  WEDNESDAY
1        reg oper account              Block  block of flats                  No      Core staff          Family                  School         Cash loans            N         State servant               Married  House / apartment               Higher education           F               N                     MONDAY
2                     NaN                NaN  

In [18]:
print(pd_loan_filter_null["AMT_INCOME_TOTAL"].value_counts().count())
pd_loan_filter_null["AMT_INCOME_TOTAL"]\
    .apply(lambda x: str(x).lower().strip()).value_counts(normalize=True)

2548


AMT_INCOME_TOTAL
135000.0       0.116256
112500.0       0.100871
157500.0       0.086358
180000.0       0.080384
90000.0        0.073113
225000.0       0.067308
202500.0       0.053140
67500.0        0.036249
270000.0       0.035208
81000.0        0.019515
315000.0       0.019063
247500.0       0.017378
126000.0       0.015180
360000.0       0.013382
121500.0       0.012513
99000.0        0.011814
292500.0       0.011115
103500.0       0.010829
76500.0        0.010231
144000.0       0.010156
171000.0       0.008608
450000.0       0.007918
117000.0       0.007772
54000.0        0.007541
72000.0        0.007112
166500.0       0.006676
45000.0        0.006644
108000.0       0.006328
337500.0       0.005821
94500.0        0.005681
162000.0       0.005535
189000.0       0.005499
405000.0       0.005398
85500.0        0.004923
148500.0       0.004832
216000.0       0.004572
130500.0       0.004084
58500.0        0.003844
63000.0        0.003723
193500.0       0.003450
153000.0       0.003291

PROCESAMIENTO INICIAL DE ALGUNAS VARIABLES

In [19]:
dia = { "MONDAY": 1, "TUESDAY": 2, "WEDNESDAY": 3, "THURSDAY": 4, "FRIDAY": 5, "SATURDAY": 6, "SUNDAY": 7}

pd_loan_filter_null['NWEEKDAY_PROCESS_START'] = pd_loan_filter_null['WEEKDAY_APPR_PROCESS_START'].replace(dia).astype("category")

C:\Users\Sara\AppData\Local\Temp\ipykernel_4804\2288118506.py:3: FutureWarning:

Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`

C:\Users\Sara\AppData\Local\Temp\ipykernel_4804\2288118506.py:3: FutureWarning:

The behavior of Series.replace (and DataFrame.replace) with CategoricalDtype is deprecated. In a future version, replace will only be used for cases that preserve the categories. To change the categories, use ser.cat.rename_categories instead.



In [20]:
pd_loan_filter_null.head()

COMMONAREA_AVG  COMMONAREA_MODE  COMMONAREA_MEDI  NONLIVINGAPARTMENTS_MEDI  NONLIVINGAPARTMENTS_MODE  NONLIVINGAPARTMENTS_AVG FONDKAPREMONT_MODE  LIVINGAPARTMENTS_AVG  LIVINGAPARTMENTS_MEDI  LIVINGAPARTMENTS_MODE  FLOORSMIN_MODE  FLOORSMIN_AVG  FLOORSMIN_MEDI  YEARS_BUILD_AVG  YEARS_BUILD_MODE  YEARS_BUILD_MEDI  OWN_CAR_AGE  LANDAREA_MEDI  LANDAREA_AVG  LANDAREA_MODE  BASEMENTAREA_MODE  BASEMENTAREA_MEDI  BASEMENTAREA_AVG  EXT_SOURCE_1  NONLIVINGAREA_MODE  NONLIVINGAREA_AVG  NONLIVINGAREA_MEDI  ELEVATORS_AVG  ELEVATORS_MEDI  ELEVATORS_MODE WALLSMATERIAL_MODE  APARTMENTS_AVG  APARTMENTS_MEDI  APARTMENTS_MODE  ENTRANCES_MODE  ENTRANCES_MEDI  ENTRANCES_AVG  LIVINGAREA_AVG  LIVINGAREA_MEDI  LIVINGAREA_MODE  HOUSETYPE_MODE  FLOORSMAX_MODE  FLOORSMAX_AVG  FLOORSMAX_MEDI  YEARS_BEGINEXPLUATATION_MODE  YEARS_BEGINEXPLUATATION_MEDI  YEARS_BEGINEXPLUATATION_AVG  TOTALAREA_MODE EMERGENCYSTATE_MODE OCCUPATION_TYPE  EXT_SOURCE_3  AMT_REQ_CREDIT_BUREAU_HOUR  AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  AMT_REQ_CREDIT_BUREAU_YEAR  AMT_REQ_CREDIT_BUREAU_DAY  AMT_REQ_CREDIT_BUREAU_QRT NAME_TYPE_SUITE  DEF_60_CNT_SOCIAL_CIRCLE  OBS_60_CNT_SOCIAL_CIRCLE  DEF_30_CNT_SOCIAL_CIRCLE  OBS_30_CNT_SOCIAL_CIRCLE  EXT_SOURCE_2  AMT_GOODS_PRICE  AMT_ANNUITY  CNT_FAM_MEMBERS  DAYS_LAST_PHONE_CHANGE  HOUR_APPR_PROCESS_START  LIVE_REGION_NOT_WORK_REGION  REG_REGION_NOT_LIVE_REGION       ORGANIZATION_TYPE NAME_CONTRACT_TYPE FLAG_OWN_CAR  SK_ID_CURR  AMT_CREDIT  AMT_INCOME_TOTAL  CNT_CHILDREN NAME_INCOME_TYPE    NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  REGION_POPULATION_RELATIVE            NAME_EDUCATION_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  DAYS_REGISTRATION  DAYS_ID_PUBLISH  FLAG_MOBIL  FLAG_EMP_PHONE  FLAG_WORK_PHONE  FLAG_CONT_MOBILE CODE_GENDER  TARGET FLAG_OWN_REALTY  FLAG_EMAIL  REGION_RATING_CLIENT  REGION_RATING_CLIENT_W_CITY WEEKDAY_APPR_PROCESS_START  FLAG_PHONE  REG_CITY_NOT_LIVE_CITY  REG_CITY_NOT_WORK_CITY  LIVE_CITY_NOT_WORK_CITY  REG_REGION_NOT_WORK_REGION  FLAG_DOCUMENT_4  FLAG_DOCUMENT_5  FLAG_DOCUMENT_2  FLAG_DOCUMENT_3  FLAG_DOCUMENT_11  FLAG_DOCUMENT_10  FLAG_DOCUMENT_9  FLAG_DOCUMENT_8  FLAG_DOCUMENT_7  FLAG_DOCUMENT_6  FLAG_DOCUMENT_12  FLAG_DOCUMENT_13  FLAG_DOCUMENT_19  FLAG_DOCUMENT_18  FLAG_DOCUMENT_17  FLAG_DOCUMENT_16  FLAG_DOCUMENT_15  FLAG_DOCUMENT_14  FLAG_DOCUMENT_20  FLAG_DOCUMENT_21 NWEEKDAY_PROCESS_START
0          0.0143           0.0144           0.0144                    0.0000                       0.0                   0.0000   reg oper account                0.0202                 0.0205                  0.022          0.1250         0.1250          0.1250           0.6192            0.6341            0.6243          NaN         0.0375        0.0369         0.0377             0.0383             0.0369            0.0369      0.083037                 0.0             0.0000                0.00           0.00            0.00          0.0000       Stone, brick          0.0247           0.0250           0.0252          0.0690          0.0690         0.0690          0.0190           0.0193           0.0198  block of flats          0.0833         0.0833          0.0833                        0.9722                        0.9722                       0.9722          0.0149                  No        Laborers      0.139376                         0.0                         0.0                        0.0                         1.0                        0.0                        0.0   Unaccompanied                       2.0                       2.0                       2.0                       2.0      0.262949         351000.0      24700.5              1.0                 -1134.0                       10                            0                           0  Business Entity Type 3         Cash loans            N      100002    406597.5          202500.0             0          Working  Single / not married  House / apartment                    0.018801  Secondary / secondary special       -9461           -637         

In [21]:
pd_loan_filter_null.shape

(307511, 123)

In [22]:
pd_loan_filter_null.to_csv("../data/pd_data_initial_preprocessing.csv")